### PR TITLE
Refactor local anchor to prevent page reload

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#6333

*Description of changes:*

* Refactor local anchor to prevent page reload

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

File: client/src/components/common/link-row.vue [hunk 1]
```diff
@@ -12,7 +12,9 @@
     <a id="name-build-link"
        class="middle-link pt-2"
        :class="{ 'no-selector-link' : !entitySelectorUrl }"
-       href="#name-build-info"
+       href="#"
+       onclick="return false;"
```
Comment (severinbeauvais) [inline client/src/components/common/link-row.vue:16]: I think you can also do `onclick=""`.

(Or onclick.prevent or onclick.stop.)

Try it.

File: client/src/components/mixins/common-mixin.ts [hunk 2]
```diff
@@ -59,4 +59,9 @@ export default class CommonMixin extends Vue {
     return [EntityType.XCR, EntityType.XUL, EntityType.RLC, EntityType.XLP, EntityType.XLL,
       EntityType.XCP, EntityType.XSO].includes(nr?.entity_type_cd)
   }
+
+  /** Scroll to given element Id */
+  private scrollTo (id) {
+    document.getElementById(id).scrollIntoView()
```
Comment (severinbeauvais) [inline client/src/components/mixins/common-mixin.ts:65]: I would add conditional chaining to avoid problems if the id isn't found.

Comment (severinbeauvais) [reply client/src/components/mixins/common-mixin.ts:65]: If possible, test this with Safari as well. I had to do the following to get scrolling working previously:
https://github.com/bcgov/business-filings-ui/blob/1c6a86ab2154646f806eb3d866bddd6beb56e32e/src/views/StandaloneDirectorsFiling.vue#L840

File: client/src/components/common/link-row.vue (hunk 1 above)
Comment (cameron-freshworks) [reply client/src/components/common/link-row.vue:16]: Kk, will do.

File: client/src/components/mixins/common-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply client/src/components/mixins/common-mixin.ts:65]: ok, good call. Will do!

File: client/src/components/common/link-row.vue (hunk 1 above)
Comment (cameron-freshworks) [reply client/src/components/common/link-row.vue:16]: I'm not entirely sure why, but neither had the same effect.
From what i read, it was one of two options that were recommended. 
`onclick="return false;"` or `href="javascript:void(0)"`

File: client/src/components/mixins/common-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply client/src/components/mixins/common-mixin.ts:65]: So it navigates to the section, but safari doesn't support SMOOTH scrolling

Comment (cameron-freshworks) [reply client/src/components/mixins/common-mixin.ts:65]: To support smooth scrolling for Safari it is either..
a) Some custom javascript
b) Add package to handle this.

File: client/src/components/common/link-row.vue (hunk 1 above)
Comment (severinbeauvais) [reply client/src/components/common/link-row.vue:16]: OK, thanks for checking.

File: client/src/components/mixins/common-mixin.ts (hunk 2 above)
Comment (severinbeauvais) [reply client/src/components/mixins/common-mixin.ts]: Ask UX / POs where to go from here.

Comment (severinbeauvais) [reply client/src/components/mixins/common-mixin.ts]: CSS transition?

**Review by severinbeauvais — APPROVED** [review]

File: client/src/components/mixins/common-mixin.ts (hunk 2 above)
Comment (cameron-freshworks) [reply client/src/components/mixins/common-mixin.ts]: Copy that, will do!

**Pull request merged**